cclib/method/ddec.py

```diff
@@ -631,67 +685,258 @@ def _calculate_w_and_u(self):
-                # Equation 69, 70, and 71
-                self._rho_A_avg[atomi] = self._spherical_average_from_cartesian(
-                    self._rho_A[atomi].data, atomi, self.radial_grid_r[atomi]
-                )
-                self._rho_A_avg[atomi] = numpy.maximum.accumulate(self._rho_A_avg[atomi][::-1])[
-                    ::-1
-                ]
-                self._theta[atomi] = self._spherical_average_from_cartesian(
-                    (1 - self._rho_cond_cartesian[atomi] / self.rho_cond.data)
-                    * self._rho_A[atomi].data,
-                    atomi,
-                    self.radial_grid_r[atomi],
-                )
-                self._theta[atomi] = numpy.maximum.accumulate(self._theta[atomi][::-1])[::-1]
-                self._wavg[atomi] = self._spherical_average_from_cartesian(
-                    self._rho_cond_cartesian[atomi] / self.rho_cond.data,
-                    atomi,
-                    self.radial_grid_r[atomi],
-                )
-                self._wavg[atomi] = numpy.maximum.accumulate(self._wavg[atomi][::-1])[::-1]
+            # Equation 69, 70, and 71
+            self._rho_A_avg[atomi] = self._spherical_average_from_cartesian(
+                self._rho_A[atomi].data, atomi, self.radial_grid_r[atomi]
+            )
+            self._rho_A_avg[atomi] = numpy.maximum.accumulate(self._rho_A_avg[atomi][::-1])[::-1]
+            self._theta[atomi] = self._spherical_average_from_cartesian(
+                (1 - self._rho_cond_cartesian[atomi] / self.rho_cond.data)
+                * self._rho_A[atomi].data,
+                atomi,
+                self.radial_grid_r[atomi],
+            )
+            self._theta[atomi] = numpy.maximum.accumulate(self._theta[atomi][::-1])[::-1]
+            self._wavg[atomi] = self._spherical_average_from_cartesian(
+                self._rho_cond_cartesian[atomi] / self.rho_cond.data,
+                atomi,
+                self.radial_grid_r[atomi],
+            )
+            self._wavg[atomi] = numpy.maximum.accumulate(self._wavg[atomi][::-1])[::-1]
             # Equation 72, 73, and 77
             N_A.append(self._rho_A[atomi].integrate())
             self.rho_wavg.append(
-                (self._theta[atomi] + self._rho_A_avg[atomi] * self._wavg[atomi] / 5)
-                / (1 - (4 / 5) * self._wavg[atomi])
+                (self._theta[atomi] + self._rho_A_avg[atomi] * self._wavg[atomi] / 5.0)
+                / (1.0 - (4.0 / 5.0) * self._wavg[atomi])
             )
-            self.u_A.append(self._integrate_from_radial([self._theta[atomi]], [atomi]))
+            u_A.append(self._integrate_from_radial([self._theta[atomi]], [atomi]))
+        self.u_A.append(u_A)
 
         return N_A
 
+    def reshape_G(self):
+        """ Calculate G_A(r_A) and reshape densities
+            [STEP 4-7]
+            TODO: some similarities exist with step 3 routine. code refactoring may improve readability.
```

Comment:
Yes, this looks to be true for the `while phiI` block at least.

Reply:
Yes. I think I'll try to factor this out after moving stockholder routines into a separate object.